setup.py

```diff
@@ -185,6 +185,7 @@ def has_ext_modules(_placeholder):
 
 extras_require = {
     "dev": [
+        "packaging>=24.0",
```

Comment:
Since we actually import this `smartsim._core._install.buildenv` (and thus `packaging`) when a user runs `smart build` can we actually make this a regular `install_requires` dependency in the `deps` list? Otherwise I think there might be a super weird corner case when a user runs `smart build` when installing without the optional `ml` dependencies.

Reply:
Can confirm that this very weird sequence of actions will lead to an error:
```sh
$ pip install .  # Very intentionally installing smartsim w/o `dev` or `ml` deps
# Abridged
Successfully built smartsim
# Abrigdged
Installing collected packages:  # Abridged

$ pip list | grep packaging
# No output is returned
# This is not a problem when installing with `dev` because we require it there
# This is not a problem when installing with `ml` because TF (and maybe scipy?) have a dependency on `packaging`

$ smart build
Traceback (most recent call last):
# Abridged
  File "/Users/mrdro/repos/ssimdev/ss/venv/lib/python3.11/site-packages/smartsim/_core/_install/buildenv.py", line 38, in <module>
    from packaging.version import InvalidVersion, Version, parse
ModuleNotFoundError: No module named 'packaging'

Reply:
Thank you, sir. Fixed!